code/modules/mob/living/living.dm

```diff
@@ -700,28 +704,46 @@
 /*////////////////////
 	RESIST SUBPROCS
 */////////////////////
+/**
+  * Checks if src can resist out of a grab.
+  *
+  * Returns FALSE (Not stunned) if not stunned or stunned by a grab, and TRUE otherwise.
+  */
+/mob/living/proc/resist_grab_check()
+	if(!IsStunned()) // If not stunned, you can resist.
+		return FALSE
+
+	for(var/I in grabbed_by) // Otherwise, if you're being stunned by a grab
+		var/obj/item/grab/G = I
+		if(G.state == GRAB_AGGRESSIVE || G.state == GRAB_NECK)
```

Comment:
with this you can spam break out of grabs again... going back full circle.
Why not just remove the stun or do a periodic stun/weaken? Or a weaken/stun on release?
This way you can still be stunned by a baton and grabbed without breaking out while stunned. And you can break out of grabs without being stunned

Reply:
> Why not just remove the stun or do a periodic stun/weaken? Or a weaken/stun on release?

There already is a weaken/stun on release, in `code/modules/mob/mob_grab.dm`. It adds those effects every life tick and they don't get reset after you get free.

> This way you can still be stunned by a baton and grabbed without breaking out while stunned.

If you're stunbatoned and grabbed passively or aggressively, you'll be able to resist like normal once the stun ends.
If you've been neck grabbed then admittedly you *are* able to resist while stunned, but getting out of the grab won't stop the stun time. So you'll just end up lying on the floor for however many seconds are left.

> And you can break out of grabs without being stunned

The stun check only applies to neck grabs, which stun you for every tick you've been grabbed for. So there'll never be a case when you can't escape except if you're being strangled, which is unbreakable by design.

Reply:
> > This way you can still be stunned by a baton and grabbed without breaking out while stunned.
> 
> If you're stunbatoned and grabbed passively or aggressively, you'll be able to resist like normal once the stun ends.
> If you've been neck grabbed then admittedly you _are_ able to resist while stunned, but getting out of the grab won't stop the stun time. So you'll just end up lying on the floor for however many seconds are left.

I just remembered stun batons also weaken somebody. As do most ways of "stunning".
Should be fine then since you check for weaken prior.